Respect the maximum receive buffer size for datagram sockets

ThFabba opened the pull request:
## Purpose

This avoids unlimited buildup of received data, leading to OOM.

JIRA issue: [CORE-14048](https://jira.reactos.org/browse/CORE-14048)

learn-more commented on drivers/network/afd/afd/bind.c:
```diff
@@ -51,13 +51,13 @@ NTSTATUS WarmSocketForBind( PAFD_FCB FCB, ULONG ShareType ) {
         }
 
-        if (NT_SUCCESS(Status))
+        if (NT_SUCCESS(Status) && FCB->Recv.Content < FCB->Recv.Size)
```
In this case we drop packets that are received last,
would it make sense to drop the first packets instead?

(I have no clue about this, merely asking to understand the reasoning behind this)

ThFabba replied:
For stream sockets, new data is "paused" (from an application standpoint) until the buffer has space again. For datagram sockets, either an existing or a new datagram has to be dropped. It was really just intuition that made me drop the new one rather than an existing one, but:
* according to a quick Google search this seems to be the expected behavior
* since there's a "peek" functionality, you can't remove the oldest if it has already been looked at, which also suggests that newer packets should be dropped.
I wish we had tests for this but apparently we don't... my patch essentially generates no difference on testbot.

learn-more replied:
Thanks for explaining.

prozacchiwawa replied:
Writing a test for this should be relatively easy given an app that receives datagrams and waits a long time between receives.  Fairly certain this is closer to correct.

ThFabba replied:
Yup I'm actually working on one already. Just have to deal with an unrelated regression (CS-14103) first.

ThFabba commented:
Alright, I've added the test. It needed some fixes to SO_RCVBUF and FIONREAD handling in order to work properly... technically those would deserve their own tests, but I'm too lazy to go that far.
Tests show that my initial approach (the simple `Content < Size` check, still receiving a full datagram) was correct, so I've adjusted the fix accordingly.

```
Running Wine Test, Module: ws2_32, Test: recv
recv.c:154: Assigned port: 32768
recv.c:160: Buffer size: 65535

recv: 65553 tests executed (0 marked as todo, 0 failures), 0 skipped.
Test recv completed in 1.11 seconds.
```
